torch/csrc/jit/passes/quantization.cpp

```diff
@@ -426,5 +486,46 @@ void FoldQuantNodesIntoInputsOutputs(std::shared_ptr<Graph>& graph) {
   throw std::runtime_error("Pass not implemented yet!");
 }
 
+void InsertQuantDequantNodesForParam(
+  script::Method& method,
+  const std::string& param_name,
+  const std::function<std::tuple<std::string, float,
+    int>(at::Tensor)>& getQParamFunc) {
+  AT_ASSERT(getQParamFunc != nullptr);
+  auto params_to_insert_qdq = getQuantizableParamsofType(method, param_name);
+
+  for (auto& param_info : params_to_insert_qdq) {
+    auto& param_slot = method.initial_ivalues()[param_info.idx];
+    const auto& itensor = param_slot.value();
+    at::Tensor tensor_var = itensor.toTensor().detach();
+    auto qparam = getQParamFunc(tensor_var);
+    addQuantDeQuantNodesForInput(param_info.v, param_info.n, qparam);
+  }
+ }
+
+ // We can have same param name apply different getQParamFunc or different
```

Comment:
Why do we need templates here? Cany you just pass a getQparamFunc as a function pointer here?

Reply:
as discussed we will retain the template for param as it will allow to keep same interface once we have per channel quant in future. Bias i will expose a separate interface because it has custom logic